### DISCUSSION TITLE
DataTable the column names is fixed #2017

### DISCUSSION THREAD

**Discussion by batiyaa** [flet-dev/flet#2018] [Q&A]
### Question

I have a listview, it has a datatable. How do I make the column names fixed so that when scrolling I can see the column names.

### Code sample

```python
table = ft.DataTable(
        columns=[
            ft.DataColumn(ft.Text("Fisrt")),
            ft.DataColumn(ft.Text("Second")),
        ],
        rows=rows,
    )
    
    listview = ft.ListView(expand=1, spacing=10, padding=20, on_scroll_interval=20)
    listview.controls.append(table)
    listview.on_scroll = on_scroll_event
```

### ------------------------------------------------------

- [x] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
https://pub.dev/packages/data_table_2 ?

**Reply by Vinny1210** [reply]
I have the same query, I don't understand why you share the flutter package, how we can use it or flet will add the same table widget in next version? thanks

**Reply by ndonkoHenri** [reply]
Wasn't that explicit, sorry.
I was asking if that package could satisfy the needs.

**Reply by Vinny1210** [reply]
that package satisfy the needs but it's for flutter not flet. The flet datatable widget not good enough. Could you please update it priority? Appreciate

**Reply by FeodorFitsner** [reply]
I like that DataTable2 widget! Let's make a new issue to implement that?

**Comment by Vinny1210** [comment]
> * [x]  I have searched for answers to my question both in the issues and in previous discussions.

May I know how did you implement the fixed column name with current datatable widget, thanks

**Reply by batiyaa** [reply]
I haven't really found a solution. For many reasons, flet does not satisfy in development even in simple projects. I decided to rewrite everything on react + Antd table. It's easier and has more features. ChatGPT will help you.
